core/src/data/des/json.cpp

```diff
@@ -247,8 +256,7 @@ bool JSONDeserializer::decompose(const Type& type, void* value)
 
         if (!mIterator->is_object())
         {
-            CUBOS_WARN("Type {} must be deserialized from a JSON object, not a {}", type.name(),
-                       mIterator->type_name());
+            CUBOS_CRITICAL("Expected an object {}, found a {}", type.name(), mIterator->type_name());
```

Comment:
```suggestion
            CUBOS_WARN("Expected an object of type {}, found a {}", type.name(), mIterator->type_name());
```
It's not a critical message - as it does not lead to a crash.